client/lib/cart-values/cart-items.js

```diff
@@ -714,15 +725,29 @@ function isDomainBeingUsedForPlan( cart, domain ) {
 	return false;
 }
 
-function shouldBundleDomainWithPlan( withPlansOnly, selectedSite, cart, suggestion ) {
+function shouldBundleDomainWithPlan( withPlansOnly, selectedSite, cart, suggestionOrCartItem ) {
 	return withPlansOnly &&
-		( suggestion.product_slug && suggestion.cost ) && // not free
-		( ! isDomainBeingUsedForPlan( cart, suggestion.domain_name ) ) && // a plan in cart
+		// not free or a cart item
+		( isDomainRegistration( suggestionOrCartItem ) ||
+			isDomainMapping( suggestionOrCartItem ) ||
+			( suggestionOrCartItem.domain_name && ! isWordPressDomain( suggestionOrCartItem ) ) ) &&
+		( ! isDomainBeingUsedForPlan( cart, suggestionOrCartItem.domain_name ) ) && // a plan in cart
 		( ! isNextDomainFree( cart ) ) && // domain credit
 		( ! hasPlan( cart ) ) && // already a plan in cart
 		( ! selectedSite || ( selectedSite && selectedSite.plan.product_slug === 'free_plan' ) ); // site has a plan
 }
 
+function bundleItemWithPlan( cartItem, planSlug = 'value_bundle' ) {
+	return [ cartItem, planItem( planSlug, false ) ];
+}
+
+function bundleItemWithPlanIfNecessary( cartItem, withPlansOnly, selectedSite, cart, planSlug ) {
```

Comment:
I think you should add a default value for `planSlug` here too. Right now you are depending on `bundleItemWithPlan`, where there's a default value (`value_bundle`) set. But if someone wants to use `planSlug` in _this_ context (`bundleItemWithPlanIfNecessary`), it will be undefined.